src/utils/data-validator.ts

```
@@ -0,0 +1,239 @@
+			.replace(/[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]/g, '') // Remove control characters
+			.replace(/javascript:/gi, '') // Remove javascript: protocol
+			.replace(/data:/gi, '') // Remove data: protocol
+			.replace(/vbscript:/gi, '') // Remove vbscript: protocol
+			.trim();
+	}
+
+	/**
+	 * Sanitizes number data
+	 */
+	private sanitizeNumber(num: number): number {
+		if (!isFinite(num)) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)